app/controllers/parties_controller.rb

```diff
@@ -1,60 +1,57 @@
 class PartiesController < ApplicationController
-
   def index
-    @parties = order_list(Party.eager_all, :name)
-
-    format({ serialized_data: @parties })
+    @parties = Parliament::Request.new.parties.get
   end
 
   def current
-    @parties = order_list(Party.eager_all('current'), :name)
-
-    format({ serialized_data: @parties })
+    @parties = Parliament::Request.new.parties.current.get
   end
 
   def show
-    @party = Party.eager_find(params[:id]) or not_found
+    id = params[:id]
+    data = Parliament::Request.new.parties(id).get
 
-    format({ serialized_data: @party })
+    @party = data.first
   end
 
   def members
-    @party = Party.eager_find(params[:party_id], 'members') or not_found
-    @members = order_list(@party.members, :surname, :forename)
+    party_id = params[:party_id]
+    data = Parliament::Request.new.parties(party_id).members.get
 
-    format({ serialized_data: @party })
+    @people = data.filter('http://id.ukpds.org/schema/Person')
   end
 
   def current_members
-    @party = Party.eager_find(params[:party_id], 'members', 'current') or not_found
-    @members = order_list(@party.members, :surname, :forename)
+    party_id = params[:party_id]
+    data = Parliament::Request.new.parties(party_id).members.current.get
 
-    format({ serialized_data: @party })
+    @people = data.filter('http://id.ukpds.org/schema/Person')
   end
 
   def letters
     letter = params[:letter]
-    @root_path = parties_a_z_path
-    @parties = order_list(Party.eager_all(letter), :name)
+    data = Parliament::Request.new.parties(letter).get
 
-    format({ serialized_data: @parties })
+    @party = data.first
   end
 
   def members_letters
     letter = params[:letter]
-    @root_path = party_members_a_z_path
-    @party = Party.eager_find(params[:party_id], 'members', letter)
-    @members = order_list(@party.members, :surname, :forename)
+    party_id = params[:party_id]
+    data_party = Parliament::Request.new.parties(party_id).get
+    data_members = Parliament::Request.new.parties(party_id).members(letter).get
 
-    format({ serialized_data: @party })
+    @party = data_party.first
+    @people = data_members.filter('http://id.ukpds.org/schema/Person')
   end
 
   def current_members_letters
     letter = params[:letter]
-    @root_path = party_members_current_a_z_path
-    @party = Party.eager_find(params[:party_id], 'members', 'current', letter)
-    @members = order_list(@party.members, :surname, :forename)
+    party_id = params[:party_id]
+    data_party = Parliament::Request.new.parties(party_id).get
+    data_current_members = Parliament::Request.new.parties(party_id).members.current(letter).get
```

Comment:
this has been done. Please see new pull request for branch: aebirim/new-changes-party-controller